src/server/database/Database/DatabaseWorkerPool.cpp

```diff
@@ -16,248 +16,428 @@
 #include <sstream>
 #endif
 
-#if MARIADB_VERSION_ID >= 100600
-#define MIN_MYSQL_SERVER_VERSION 100500u
-#define MIN_MYSQL_CLIENT_VERSION 30203u
+#if defined(LIBMARIADB) && MARIADB_VERSION_ID >= 100600
+#define MIN_DB_SERVER_VERSION 100500u
+#define MIN_DB_CLIENT_VERSION 30203u
 #else
-#define MIN_MYSQL_SERVER_VERSION 50700u
-#define MIN_MYSQL_CLIENT_VERSION 50700u
+#define MIN_DB_SERVER_VERSION 50700u
+#define MIN_DB_CLIENT_VERSION 50700u
 #endif
 
-class PingOperation : public SQLOperation
+constexpr auto MAX_SYNC_CONNECTIONS = 32;
+constexpr auto MAX_ASYNC_CONNECTIONS = 32;
+
+class PingOperation : public AsyncOperation
 {
-    //! Operation for idle delaythreads
-    bool Execute() override
+public:
+    explicit PingOperation() :
+        AsyncOperation() { }
+
+    //! Operation for idle delay threads
+    void ExecuteQuery() override
     {
-        m_conn->Ping();
-        return true;
+        _connection->Ping();
     }
 };
 
-template <class T>
-DatabaseWorkerPool<T>::DatabaseWorkerPool() :
-    _queue(new ProducerConsumerQueue<SQLOperation*>()),
-    _async_threads(0),
-    _synch_threads(0)
+DatabaseWorkerPool::DatabaseWorkerPool(DatabaseType type) :
+    _poolType(type)
 {
-    WPFatal(mysql_thread_safe(), "Used MySQL library isn't thread-safe.");
+    ASSERT(mysql_thread_safe(), "Used MySQL library isn't thread-safe");
 
-#if !defined(MARIADB_VERSION_ID) || MARIADB_VERSION_ID < 100600
-    bool isSupportClientDB = mysql_get_client_version() >= MIN_MYSQL_CLIENT_VERSION;
+#if !defined(LIBMARIADB) && MARIADB_VERSION_ID >= 100600
+    bool isSupportClientDB = mysql_get_client_version() >= MIN_DB_CLIENT_VERSION;
     bool isSameClientDB = mysql_get_client_version() == MYSQL_VERSION_ID;
 #else // MariaDB 10.6+
-    bool isSupportClientDB = mysql_get_client_version() >= MIN_MYSQL_CLIENT_VERSION;
+    bool isSupportClientDB = mysql_get_client_version() >= MIN_DB_CLIENT_VERSION;
     bool isSameClientDB    = true; // Client version 3.2.3?
 #endif
 
-    WPFatal(isSupportClientDB, "AzerothCore does not support MySQL versions below 5.7 or MariaDB versions below 10.5.\n\nFound version: {} / {}. Server compiled with: {}.\nSearch the wiki for ACE00043 in Common Errors (https://www.azerothcore.org/wiki/common-errors#ace00043).",
-        mysql_get_client_info(), mysql_get_client_version(), MYSQL_VERSION_ID);
-    WPFatal(isSameClientDB, "Used MySQL library version ({} id {}) does not match the version id used to compile AzerothCore (id {}).\nSearch the wiki for ACE00046 in Common Errors (https://www.azerothcore.org/wiki/common-errors#ace00046).",
-        mysql_get_client_info(), mysql_get_client_version(), MYSQL_VERSION_ID);
+    ASSERT(isSupportClientDB, "AzerothCore does not support MySQL versions below 5.7 and MariaDB 10.3");
+    ASSERT(isSameClientDB, "Used DB library version ({} id {}) does not match the version id used to compile AzerothCore (id {})", mysql_get_client_info(), mysql_get_client_version(), MYSQL_VERSION_ID);
```

Comment:
\nSearch the wiki for ACE00043 in Common Errors (https://www.azerothcore.org/wiki/common-errors#ace00043)
\nSearch the wiki for ACE00046 in Common Errors (https://www.azerothcore.org/wiki/common-errors#ace00046).

Reply:
Ah yes, thanks

Reply:
Added